### ISSUE TITLE
Random silent crashes

### ISSUE THREAD

**Issue by TheInfamousAlk** [bitgapp/Cursr#80]
### Describe the Bug/Issue
The program will silently crash at some point after being online for a few hours. Almost once a day.

### Your Environment
 OS: Pop!_OS 22.04 LTS x86_64 
Kernel: 6.9.3-76060903-generic
Resolution: 1200x1920, 1920x1080, 2560x1440, 3840x2160 
DE: GNOME 42.9 
WM: Mutter 
CPU: Intel i9-9900KF (16) @ 5.000GHz 
GPU: NVIDIA GeForce RTX 2080 Ti Rev. A

Cursr version 1.7.2

Latest log attached.
[app.log](https://github.com/user-attachments/files/17138485/app.log)

**Comment by TheInfamousAlk** [comment]
Here's my monitor configuration if this matters: 
![image](https://github.com/user-attachments/assets/640ab7d7-7cfc-44e6-9710-757181610a8f)

**Comment by ragauskl** [comment]
I downloaded the log file and removed it, I prefer to not keep individual user logs in public place like github.

I do see compatibility issues, first thing is Xrandr is missing, you can try installing it and see if that will resolve the issue. Another issue is 'Error reading path of process from /proc/0/exe' and similar which I haven't seen before at all, unless I do some testing on Pop!_OS I can't say for sure what the underlaying issue may be as it's first reported occurrence. Both errors may or may not be related to the actual segmentation fault crash, but all could be due to Pop!_OS compatibility.

I noted down to look into this but right now can't promise a time estimate/when to expect a fix as right now there are higher priority tasks on my list. I'll provide an update when I'll start looking into this

**Comment by TheInfamousAlk** [comment]
Okay thank you for removing that.

It looks like on my repo, xrandr doesn't exist, and it says `x11-xserver-utils` replaces it. I will try installing that, or do you recommend I manually find and install Xrandr instead? Shown below, neither is listed in the dependencies for apt. 

```
~$ apt-cache depends cursr
cursr
  Depends: libgtk-3-0
  Depends: libnotify4
  Depends: libnss3
  Depends: libxss1
  Depends: libxtst6
  Depends: xdg-utils
  Depends: libatspi2.0-0
  Depends: libuuid1
  Depends: libsecret-1-0
  Recommends: libappindicator3-1
    libayatana-appindicator3-1
```

I don't know anything about the /proc/0/exe thing so I can't give any insight on that unless you have specific questions for me.

**Comment by ragauskl** [comment]
You can try installing x11-server-utils, if the errors 'Xrandr missing' disappear from logs - then you'll know it worked, if you still see the errors then I'd need to look into alternative solution.

Regarding '/proc/0/exe ' the process number does not matter, my guess it's either Cursr not being able to read data about it's own processes or Cursr not being able to get data on other processes when it checks for any focused windows where 'auto-disable' should get triggered (if you have set auto-disable) on any of the processes. 

I think it's more likely that xrandr issue is related to the crash than the '/proc/0/exe' one.

**Comment by ragauskl** [comment]
Before you look into Xrandr, can you delete old log files, enable Debug mode in settings and wait for the crash to happen, then send me the logs files (all app.log, like app2.log etc if there are any) to contact@cursr.app.

As long as the issue does not occur you can delete the logs as often as you'd like in case you're concerned about key strokes being included in the log from extended period of time.

Then I may better understand if the crash happens when functions using xrandr are being called or is it something else - in which case Xrandr issue is small inconvenience but not the cause of the crash.

**Comment by TheInfamousAlk** [comment]
Yeah, I will wait a couple days for it to crash again. I just deleted my logs and turned on Debug mode. Thanks!

**Comment by TheInfamousAlk** [comment]
So my power went out for 3 days as I got hit with hurricane Helene, but now I'm back! And it has happened again, but it seems like debug mode has been disabled without my input since I last enabled it, and no extra data was captured in the logs. I will now start keeping an eye on debug mode and enabling it manually when I notice it has been disabled, at least until we get the logs we need.

**Comment by ragauskl** [comment]
Oh wow, that sounds like tough time, hope you're well and won't get much damage from it.  As for the issue, not sure why it would get disabled as this can happen only if something goes wrong in app during loading  time and then all settings can get reset including setups - but you'd notice that and I only saw that in my local development environment. Another time is I think there was a version going from beta to stable where I may have added a reset for enforced debug mode, but that's been removed since, and unlikely you had an update from an old version. If this keeps happening let me know

**Comment by TheInfamousAlk** [comment]
Alright it has crashed twice today, I emailed the logs to the email you mentioned above. Let me know if you need any more info!

**Comment by TheInfamousAlk** [comment]
It might actually have to do with me alt tabbing from overwatch. It just happened again, so that is at least 2 times today while playing overwatch. I have it in exclusive fullscreen (not borderless) mode. Can't say for sure though.
